packages/calling-component-bindings/src/hooks/usePropsFor.ts

```diff
@@ -80,28 +80,17 @@ const emptySelector: EmptySelector = (): Record<string, never> => ({});
  *
  * @public
  */
-export type GetSelector<Component extends (props: any) => JSX.Element | undefined> = AreEqual<
-  Component,
-  typeof VideoGallery
-> extends true
-  ? VideoGallerySelector
-  : AreEqual<Component, typeof DevicesButton> extends true
-  ? DevicesButtonSelector
-  : AreEqual<Component, typeof MicrophoneButton> extends true
-  ? MicrophoneButtonSelector
-  : AreEqual<Component, typeof CameraButton> extends true
-  ? CameraButtonSelector
-  : AreEqual<Component, typeof ScreenShareButton> extends true
-  ? ScreenShareButtonSelector
-  : AreEqual<Component, typeof ParticipantList> extends true
-  ? ParticipantListSelector
-  : AreEqual<Component, typeof ParticipantsButton> extends true
-  ? ParticipantsButtonSelector
-  : AreEqual<Component, typeof EndCallButton> extends true
-  ? EmptySelector
-  : AreEqual<Component, typeof ErrorBar> extends true
-  ? ErrorBarSelector
-  : undefined;
+export type GetSelector<Component extends (props: any) => JSX.Element | undefined> = FilterUndefined<
+  | (AreEqual<Component, typeof VideoGallery> extends true ? VideoGallerySelector : undefined)
```

Comment:
There could be a very small risk here that, if one selector is the super set of another (I guess EndCallButton could be one of them), there will be 2 selectors chosen in the end and this will be selectorA | selectorB

Reply:
Hence the TypeAssertions. This isn't happening in practice... yet.
If this happens, we might have to find a way to "pick the first one".

Reply:
Talked IRL and double checked that our type assertions cover this case.